### DISCUSSION TITLE
spark cluster composition 스파크 클러스터 구성

### DISCUSSION THREAD

**Discussion by SOONSOO7** [minhyeok0328/next-loop#61] [General]
![스크린샷 2024-11-05 145710](https://github.com/user-attachments/assets/8c25e454-1f62-4f56-9e59-f2626fb022b6)

<img width="511" alt="20241105_150133" src="https://github.com/user-attachments/assets/7f3dade7-f808-4231-bf5b-04ed44384c78">

[project-structure.txt](https://github.com/user-attachments/files/17627607/project-structure.txt)

윤수님, 그리고 민혁님 수진님
이렇게 파이프라인과
파이프라인 스트럭쳐를 보고
스파크 클러스터 구축을 하려고 합니다.

이야기 언제든지 말씀 주세요

---

파이프라인 수정됨. 간략하게.
각 구성 이유를 명확히 하고 확실하게.

---

gcloud compute ssh dowhat-spark-**

git clone https://github.com/minhyeok0328/next-loop.git

cd next-loop

git fetch origin

git checkout 25-issue

cd spark-cluster

ls

sudo apt-get update

sudo apt-get install -y docker.io

sudo systemctl start docker

sudo systemctl enable docker

sudo usermod -aG docker $USER

newgrp docker

groups

docker-compose up -d

---

gcloud compute firewall-rules create spark-master \
    --allow tcp:4040,tcp:7077,tcp:8080 \
    --description="Spark master node ports" \
    --direction=INGRESS

gcloud compute firewall-rules create spark-worker \
    --allow tcp:8081-8082 \
    --description="Spark worker node ports" \
    --direction=INGRESS

**Comment by SOONSOO7** [comment]
다른 폴더에서
git clone {URL}
cd next-loop
git fetch origin
git checkout 25-issue
그 다음에
.git 파일을 현재 작업중인 폴더로 옮기기

cp -r .git ~/next-loop

**Comment by dnlpys** [comment]
LGTM!!!👍

**Reply by SOONSOO7** [reply]
감사합니다!! 하나씩 하나씩 해나가고 있습니다
